Add mdi image documentation (#2837)
OttoWinter/esphomedocs: components/display/index.rst

        id: my_image
        resize: 100x100

.. code-block:: yaml

    image:
      - file: mdi:alert-outline
        id: alert
        resize: 80x80

Configuration variables:

> - **file** (**Required**, string): The path (relative to where the .yaml file is) of the image file. Instead of a local file, you can specify a *Material Design Icon* id in the format ``mdi:icon-name``, and that icon will automatically be downloaded and added to the configuration.

Reviewer: Suggest adding link to either the MDI homepage or index of icons, since some people either won't know what MDI is or won't know how to find the list of supported icons (and even if they do, it's good to reduce friction/be helpful).

Reviewer 2: I think this text is too long and could be broken up into a list something like this:
```suggestion
- **file** (**Required**, string): 
  
  - **Local files**: The path (relative to where the .yaml file is) of the image file. 
  - **Material Design Icons**: Specify the *Material Design Icon* id in the format ``mdi:icon-name``, and that icon will automatically be downloaded and added to the configuration.
```

Author: @Unfocused I've added the link to the same page that HomeAssistant is referencing; I think it's good to keep that consistent.
@jesserockz Your suggestion makes sense. I've taken it in.
